### DISCUSSION TITLE
Downloading Trash files and folders

### DISCUSSION THREAD

**Discussion by samrajcse** [box/box-java-sdk#1028] [Q&A] [answered]
Hi Team,
I am using JAVA Box SDK. I want to iterate all the files and folders in the trash with pagiantion. I don't want all the files in the single call. How can we do it?

In the sample code, pagination is not there.  I need to download the files inside the folders inside the trash as well.

**Comment by antusus** [comment]
Hi @samrajcse,

Thanks for submitting this Issue! The SDK is not allowing  you to specify limit when iterating over elements in the trash. We will take a look and get back to you ASAP!

@antusus

**Comment by antusus** [comment, marked as answer]
The Box [Java SDK 3.4.0 ](https://github.com/box/box-java-sdk/releases/tag/v3.4.0)now supports pagination while getting the trash items: https://github.com/box/box-java-sdk/blob/main/doc/trash.md#get-trashed-items

**Reply by samrajcse** [reply]
@antusus I can able to download files from trash and can't able to get the list of files in trashed folder to download the files for ediscovery purpose. Do we have any api to list down the files in the trashed folder? We can't restore back and download the file as that user is under legalhold.

**Reply by antusus** [reply]
I'm not aware of such API. Maybe you can reach out to support:
https://support.box.com/hc/en-us/requests/new
and they can offer some solution. Maybe taking all the privileges from this user and changing his password would allow you to activate this account and then browse trash. But I'm not aware of what you can do while this user is under legal hold.

There is Developer Forum which you can use to request or vote on features:
https://support.box.com/hc/en-us/community/topics/360001932973-Platform-and-Developer-Forum

Soon Box Works 2022 (October 6-7) starts and there should be a possibility to ask an expert as well:
https://boxworks.box.com/
